src/components/cart/cart.tsx

```diff
@@ -1,66 +1,84 @@
 import './cart.scss';
 import { useState } from 'react';
 import cart from '../../assets/images/icon-cart.svg';
-import { addCartItem, addToCart, removeFromCart } from '../../slices/cart';
-import { RootState } from '../../store';
-import { useSelector, useDispatch } from 'react-redux';
-import deleteIcon from '../../assets/images/icon-delete.svg';
+import { removeFromCart } from '../../slices/cart';
+import { checkoutModalDisplay } from '../../slices/modal';
+import { RootState } from '../../store'
+import { useSelector, useDispatch} from 'react-redux';
+import deleteIcon from '../../assets/images/icon-delete.svg'
 
 const Cart = () => {
-  const [showCartInfo, setShowCartInfo] = useState(false);
-  const dispatch = useDispatch();
-  const { cartItems } = useSelector((state: RootState) => state.cart);
-  const cartInfo = (e: any) => {
-    e.stopPropagation();
-    setShowCartInfo(!showCartInfo);
-  };
-  const itemAmount = (a: number, b: number): number => {
-    return a * b;
-  };
-  const removeCartItem = (id: number) => {
-    dispatch(removeFromCart(id));
-  };
-  return (
-    <div
-      // onMouseEnter={cartInfo}
-      //      onMouseLeave={cartInfo}
-      onClick={cartInfo}>
-      <div className="cart-item">
-        {cartItems.length > 0 && <div className="item-number">{cartItems.length}</div>}
-        <img src={cart} alt="your cart" className="mr-1" />
-      </div>
-      {showCartInfo && (
-        <div onClick={(e) => e.stopPropagation()}>
-          <div className="cart-info-wrapper">
+    const [showCartInfo, setShowCartInfo] = useState(false);
+    const dispatch = useDispatch();
+    const { cartItems } = useSelector((state: RootState) => state.cart);
+
+    const cartInfo = (e:any) => {
+        e.stopPropagation();
+        setShowCartInfo(!showCartInfo)
+    }
+    const itemAmount = (a:number,b: number): number => {
+        return a * b;
+    }
+
+    console.log({cartItems});
+    const removeCartItem = (id: number) => {
+        dispatch(removeFromCart(id))
+    }
+
+    const showCheckoutModal = () => {
+        console.log({checkoutModalDisplay})
+        dispatch(checkoutModalDisplay());
+    }
+
+
+    return (
+    <div 
+    // onMouseEnter={cartInfo}
+    //      onMouseLeave={cartInfo}
+    onClick={cartInfo}
+    >
+        <div className='cart-item'>
+            {
+                cartItems.length > 0 &&
+                <div className="item-number">{cartItems.length}</div>
+            }
+            <img src={cart} alt='your cart' className='mr-1'/>
+        </div>
+        {showCartInfo &&
+        <div onClick={(e) => e.stopPropagation() }>            
```

Comment:
Why are we stopping propagation?  @onyi2020

Reply:
Would have to check to be sure why again. But I am most likely keeping the click action to only happen on the parent elemnet and not triggered when the child is clicked accidentally